### DISCUSSION TITLE
Modifying library for Double HUB-75

### DISCUSSION THREAD

**Discussion by harissutanrafiq** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#265] [Ideas]
I try modified your library fo support double hub 75, hub75-1 and hub 75-2
like hiudu controller, split using diferent lat pin

HUB 75-1 LAT_PIN 22
HUB 75-2 LAT_PIN 26

![61wFwdyYb2L _SL1092_](https://user-images.githubusercontent.com/33673320/157202834-f8466718-77bc-4d78-adaa-e82e386703ac.jpg)

First i change this line
`    i2s_parallel_config_t cfg = {
        .gpio_bus={_cfg.gpio.r1, _cfg.gpio.g1, _cfg.gpio.b1, _cfg.gpio.r2, _cfg.gpio.g2, _cfg.gpio.b2, _cfg.gpio.lat, _cfg.gpio.oe, _cfg.gpio.a, _cfg.gpio.b, _cfg.gpio.c, _cfg.gpio.d, _cfg.gpio.e, _cfg.gpio.lat2, -1, -1,-1},
        .gpio_clk=_cfg.gpio.clk,
        .sample_rate=_cfg.i2sspeed,   
        .sample_width=ESP32_I2S_DMA_MODE,        
        .desccount_a=desccount,
        .lldesc_a=dmadesc_a,        
        .desccount_b=desccount,
        .lldesc_b=dmadesc_b,
        .clkphase=_cfg.clkphase
    };
`

And i added new define variable
#define BIT_LAT_2 (1 << 13)

And change MatrixPanel_I2S_DMA::clearFrameBuffer(bool _buff_id) line 640
 row[dma_buff.rowBits[row_idx]->width - 2] |= BIT_LAT_2;
 
 But thereis noise on second row

![275034432_1140428333389342_8911971564363934093_n](https://user-images.githubusercontent.com/33673320/157205361-f3bf9f6a-7e6f-4222-9fc7-b7e92d5ada6c.jpg)

any idea ..?

**Comment by tsctrl** [comment]
hi @harissutanrafiq , wondering, what is the reason/benefit/possible enhancement or control in the program by spliting to multiple hub? it cost a pin/lat.

**Reply by harissutanrafiq** [reply]
for example config is 6x2 module p5, 
if we use virtualDisplay 
 iam write string at 0, 0 (x,y);
when iam remove 2 module and config still 6x2 , the teks will not at that position

**Comment by harissutanrafiq** [comment]
how i can identified or split a data is for upper (rows) or for lower (rows) if i use 128 x 64 pixel..?
